### DISCUSSION TITLE
Search for lyrics or samples for currently playing spotify song

### DISCUSSION THREAD

**Discussion by MarianoAmado** [johnlindquist/kit#908] [Share]
![Oct-08-2022 13-36-08](https://user-images.githubusercontent.com/28148619/194717999-c0697518-3e21-40da-b292-c40e1301c743.gif)

After using and playing around with @jacopo-degattis' [spotify-playing](https://github.com/johnlindquist/kit/discussions/866), I came up this quick and simple way to look up lyrics and samples (and maybe more?) for the song that's currently playing in spotify.

I plan on adding some basic feedback on errors and probably an option for manual search input.
Contributions and suggestions always welcome!

----------

[Open spotify-searches in Script Kit](https://scriptkit.com/api/new?name=spotify-searches&url=https://gist.githubusercontent.com/MarianoAmado/376ce67943f0a228d7ea12b0973bf781/raw/fef684eb46558a63696e0bf33ce58da442ab7b37/spotify-searches.ts")

```js
// Name: Search current song
// Description: Search the currently playing spotify song for lyrics or samples
// Author: Mariano Amado

import "@johnlindquist/kit"

const spotify = await npm("spotify-node-applescript")

spotify.isRunning((err, isRunning) => {
    if (!isRunning) return exit();

    spotify.getTrack(async (err: any, track: any) => {
        if (err) return exit();
        const { artist, name } = track;

        const term = `${artist}+${name}`.replace("&", "");

        const site = await arg(`${name} - ${artist}`, [
            {
                name: "Genius.com",
                description: 'Search for song lyrics',
                value: 'genius',
            },
            {
                name: "Whosampled.com",
                description: 'Search for samples',
                value: 'whosampled',
            },
        ])

        const uri = encodeURI(`https://www.${site}.com/search?q=${term}`)
        exec(`open '${uri}'`)
    })
})

```

**Comment by johnlindquist** [comment]
Love it! Thanks for sharing 🏆

Might make sense to add these as built-in links to the widget 🤔

**Reply by MarianoAmado** [reply]
Yeah!
could be handy to add shortcuts to the widget like these, would also be really easy to customize and add new ways of searching for stuff related to the song/album

Thanks!! 🙌
